control-plane/subcommand/sync-catalog/command.go

```diff
@@ -363,12 +378,9 @@ func (c *Command) Run(args []string) int {
 	}
 }
 
-func (c *Command) handleReady(rw http.ResponseWriter, req *http.Request) {
-	// The main readiness check is whether sync can talk to
-	// the consul cluster, in this case querying for the leader
-	_, err := c.consulClient.Status().Leader()
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("[GET /health/ready] Error getting leader status: %s", err))
+func (c *Command) handleReady(rw http.ResponseWriter, _ *http.Request) {
+	if !c.ready {
+		c.UI.Error("[GET /health/ready] sync catalog controller is not yet ready")
```

Comment:
The readiness endpoint doesn't need to talk to the servers for their status. Instead we'll rely on the consul-server-connection-manager to tell us if consul servers are ready